### ISSUE TITLE
configuration unsuccessfull

### ISSUE THREAD

**Issue by aruncgowda** [dyninst/dyninst#774]
@hainest 
i am trying this from last week but i am unsuccessfull .please help me to install dyninst

- The C compiler identification is GNU 7.5.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- -- Input platform: x86_64-unknown-linux2.4
-- Set arch and platform based definitions
-- Options set
-- Set optimization flags
-- Check if the system is big endian
-- Searching 16 bit integer
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of unsigned short
-- Check size of unsigned short - done
-- Searching 16 bit integer - Using unsigned short
-- Check if the system is big endian - little endian
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/cc
-- dyninstAPI RT library SOVERSION: 10.1
-- dyninstAPI RT library LIBVERSION: 10.1.0
-- Checking for 32-bit runtime library...
-- Performing Test CHECK_RT_LIB_32
-- Performing Test CHECK_RT_LIB_32 - Success
-- Disabling 32-bit runtime library; change BUILD_RTLIB_32 to ON and install 32-bit build environment to enable
-- Configuring done
-- Generating done
-- Build files have been written to: /home/sharma/dyninst/dyninstAPI_RT
-- RTlib Makefile: 
-- Could NOT find LATEX (missing: LATEX_COMPILER) 
-- LaTeX not found
-- Adding Unix-specific dependencies
-- Configuring incomplete, errors occurred!
See also "/home/sharma/dyninst/CMakeFiles/CMakeOutput.log".
See also "/home/sharma/dyninst/CMakeFiles/CMakeError.log".

**Comment by aruncgowda** [comment]
@hainest 
this is my cmakeoutput.log

Performing C SOURCE FILE Test CMAKE_HAVE_LIBC_PTHREAD failed with the following output:
Change Dir: /home/sharma/dyninst/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/make cmTC_1ee27/fast && /usr/bin/make  -f CMakeFiles/cmTC_1ee27.dir/build.make CMakeFiles/cmTC_1ee27.dir/build
make[1]: Entering directory '/home/sharma/dyninst/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_1ee27.dir/src.c.o
/usr/bin/cc   -DCMAKE_HAVE_LIBC_PTHREAD -o CMakeFiles/cmTC_1ee27.dir/src.c.o -c /home/sharma/dyninst/CMakeFiles/CMakeTmp/src.c
Linking C executable cmTC_1ee27
/usr/local/bin/cmake -E cmake_link_script CMakeFiles/cmTC_1ee27.dir/link.txt --verbose=1
/usr/bin/cc  -DCMAKE_HAVE_LIBC_PTHREAD CMakeFiles/cmTC_1ee27.dir/src.c.o -o cmTC_1ee27 
CMakeFiles/cmTC_1ee27.dir/src.c.o: In function `main':
src.c:(.text+0x3e): undefined reference to `pthread_create'
src.c:(.text+0x4a): undefined reference to `pthread_detach'
src.c:(.text+0x56): undefined reference to `pthread_cancel'
src.c:(.text+0x67): undefined reference to `pthread_join'
src.c:(.text+0x7b): undefined reference to `pthread_atfork'
collect2: error: ld returned 1 exit status
CMakeFiles/cmTC_1ee27.dir/build.make:105: recipe for target 'cmTC_1ee27' failed
make[1]: *** [cmTC_1ee27] Error 1
make[1]: Leaving directory '/home/sharma/dyninst/CMakeFiles/CMakeTmp'
Makefile:140: recipe for target 'cmTC_1ee27/fast' failed
make: *** [cmTC_1ee27/fast] Error 2


Source file was:
#include <pthread.h>

void* test_func(void* data)
{
  return data;
}

int main(void)
{
  pthread_t thread;
  pthread_create(&thread, NULL, test_func, NULL);
  pthread_detach(thread);
  pthread_cancel(thread);
  pthread_join(thread, NULL);
  pthread_atfork(NULL, NULL, NULL);
  pthread_exit(NULL);

  return 0;
}

Determining if the function pthread_create exists in the pthreads failed with the following output:
Change Dir: /home/sharma/dyninst/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/make cmTC_ede6c/fast && /usr/bin/make  -f CMakeFiles/cmTC_ede6c.dir/build.make CMakeFiles/cmTC_ede6c.dir/build
make[1]: Entering directory '/home/sharma/dyninst/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_ede6c.dir/CheckFunctionExists.c.o
/usr/bin/cc   -DCHECK_FUNCTION_EXISTS=pthread_create -o CMakeFiles/cmTC_ede6c.dir/CheckFunctionExists.c.o -c /usr/local/share/cmake-3.18/Modules/CheckFunctionExists.c
Linking C executable cmTC_ede6c
/usr/local/bin/cmake -E cmake_link_script CMakeFiles/cmTC_ede6c.dir/link.txt --verbose=1
/usr/bin/cc  -DCHECK_FUNCTION_EXISTS=pthread_create CMakeFiles/cmTC_ede6c.dir/CheckFunctionExists.c.o -o cmTC_ede6c  -lpthreads 
/usr/bin/ld: cannot find -lpthreads
collect2: error: ld returned 1 exit status
CMakeFiles/cmTC_ede6c.dir/build.make:105: recipe for target 'cmTC_ede6c' failed
make[1]: *** [cmTC_ede6c] Error 1
make[1]: Leaving directory '/home/sharma/dyninst/CMakeFiles/CMakeTmp'
Makefile:140: recipe for target 'cmTC_ede6c/fast' failed
make: *** [cmTC_ede6c/fast] Error 2

**Comment by aruncgowda** [comment]
@hainest 
i used this commands to install dyninst

sudo apt-get update
sudo apt-get install zlib1g-dev
sudo apt-get install libpng-dev
sudo apt-get install libboost-all-dev

git clone https://github.com/dyninst/dyninst.git
cd dyninst
git checkout -b v9.3.2


cmake /home/yourusername/dyninst -DCMAKE_INSTALL_PREFIX=$HOME/.local
make install


i have been stucked from one week .please say the command to install dyninst

**Comment by bbiiggppiigg** [comment]
@aruncgowda 
Can you please install libpthread-stubs0-dev and see if this help?
And if you run into some latex related problem, try to install textlive-recommended
I think m4 is also required if you haven't install it.

**Comment by aruncgowda** [comment]
@bbiiggppiigg 
thank you
i used this commad
cd dyninst
cmake  /home/sharma/dyninst


i am getting this warnings n errors

-- Boost version: 1.65.1
-- Found the following Boost libraries:
--   atomic
--   chrono
--   date_time
--   filesystem
--   system
--   thread
--   timer
-- Boost includes: /usr/include
-- Boost library dirs: /usr/lib/x86_64-linux-gnu
-- Boost thread library: /usr/lib/x86_64-linux-gnu/libboost_thread.so
-- Boost libraries: /usr/lib/x86_64-linux-gnu/libboost_atomic.so;/usr/lib/x86_64-linux-gnu/libboost_chrono.so;/usr/lib/x86_64-linux-gnu/libboost_date_time.so;/usr/lib/x86_64-linux-gnu/libboost_filesystem.so;/usr/lib/x86_64-linux-gnu/libboost_system.so;/usr/lib/x86_64-linux-gnu/libboost_thread.so;-lpthread;/usr/lib/x86_64-linux-gnu/libboost_timer.so;-lpthread
CMake Error at cmake/Modules/FindTBB.cmake:154 (file):
  file failed to open for reading (No such file or directory):

    /usr/local/include/tbb/tbb_stddef.h
Call Stack (most recent call first):
  cmake/ThreadingBuildingBlocks.cmake:75 (find_package)
  CMakeLists.txt:23 (include)


-- Searching for tbb_preview...
-- Hints: /usr/lib /usr/local
-- Searching for tbbmalloc...
-- Hints: /usr/lib /usr/local
-- Found /usr/local/lib/libtbbmalloc.so
-- Found /usr/local/lib/libtbbmalloc.so
-- Searching for tbb...
-- Hints: /usr/lib /usr/local
-- Found /usr/local/lib/libtbb.so
-- Found /usr/local/lib/libtbb.so
-- Could NOT find TBB: Found unsuitable version ".", but required is at least "2018.6" (found /usr/local/include)
-- 
-- Attempting to build TBB(2018.6) as external project
-- TBB include directory: /usr/local/include
-- TBB library directory: /usr/local/lib
-- TBB libraries: /usr/local/lib/libtbb.so;/usr/local/lib/libtbbmalloc.so;/usr/local/lib/libtbbmalloc_proxy.so
-- TBB definitions: 
-- Could NOT find LibElf: Found unsuitable version "0.170", but required is at least "0.173" (found /usr/lib/x86_64-linux-gnu/libelf.so.1)
-- 
-- Attempting to build elfutils(0.176) as external project
-- ElfUtils includes: /usr/local/include;/usr/local/include/elfutils
-- ElfUtils library dirs: /usr/local/lib;/usr/local/lib/elfutils
-- ElfUtils libraries: /usr/local/lib/libelf.so;/usr/local/lib/libdw.so
-- -- Input platform: 
-- -- Attempting to automatically identify platform: x86_64-unknown-linux2.4
-- C++11 support found, required flags are: -std=c++11
-- -- Enabling ThreadDB support
-- Set arch and platform based definitions
-- Found g++, enabling -fvisibility=hidden
-- Options set
-- Set optimization flags
-- Building common...
-- Building dynElf...
-- Processing dependent target /usr/lib/x86_64-linux-gnu/libelf.so.1...
-- Building dynDwarf...
-- Processing dependent target dynElf...
-- Found dependency location /home/sharma/dyninst/elf
-- Processing dependent target common...
-- Found dependency location /home/sharma/dyninst/common
-- Building symLite...
-- Processing dependent target common...
-- Found dependency location /home/sharma/dyninst/common
-- Processing dependent target dynElf...
-- Found dependency location /home/sharma/dyninst/elf
-- Building instructionAPI...
-- Processing dependent target common...
-- Found dependency location /home/sharma/dyninst/common
-- Building symtabAPI...
-- Processing dependent target common...
-- Found dependency location /home/sharma/dyninst/common
-- Processing dependent target dynElf...
-- Found dependency location /home/sharma/dyninst/elf
-- Processing dependent target dynDwarf...
-- Found dependency location /home/sharma/dyninst/dwarf
-- Processing dependent target /usr/local/lib/libelf.so...
-- Processing dependent target /usr/local/lib/libdw.so...
-- Building parseAPI...
-- Processing dependent target common...
-- Found dependency location /home/sharma/dyninst/common
-- Processing dependent target instructionAPI...
-- Found dependency location /home/sharma/dyninst/instructionAPI
-- Processing dependent target symtabAPI...
-- Found dependency location /home/sharma/dyninst/symtabAPI
-- Architecture is: x86_64-linux-gnu
-- Building pcontrol...
-- Processing dependent target common...
-- Found dependency location /home/sharma/dyninst/common
-- Processing dependent target symtabAPI...
-- Found dependency location /home/sharma/dyninst/symtabAPI
-- Building stackwalk...
-- Processing dependent target common...
-- Found dependency location /home/sharma/dyninst/common
-- Processing dependent target instructionAPI...
-- Found dependency location /home/sharma/dyninst/instructionAPI
-- Processing dependent target pcontrol...
-- Found dependency location /home/sharma/dyninst/proccontrol
-- Processing dependent target dynDwarf...
-- Found dependency location /home/sharma/dyninst/dwarf
-- Processing dependent target dynElf...
-- Found dependency location /home/sharma/dyninst/elf
-- Processing dependent target symtabAPI...
-- Found dependency location /home/sharma/dyninst/symtabAPI
-- Processing dependent target parseAPI...
-- Found dependency location /home/sharma/dyninst/parseAPI
-- Building patchAPI...
-- Processing dependent target common...
-- Found dependency location /home/sharma/dyninst/common
-- Processing dependent target instructionAPI...
-- Found dependency location /home/sharma/dyninst/instructionAPI
-- Processing dependent target parseAPI...
-- Found dependency location /home/sharma/dyninst/parseAPI
-- Building dyninstAPI...
-- Processing dependent target common...
-- Found dependency location /home/sharma/dyninst/common
-- Processing dependent target instructionAPI...
-- Found dependency location /home/sharma/dyninst/instructionAPI
-- Processing dependent target stackwalk...
-- Found dependency location /home/sharma/dyninst/stackwalk
-- Processing dependent target pcontrol...
-- Found dependency location /home/sharma/dyninst/proccontrol
-- Processing dependent target patchAPI...
-- Found dependency location /home/sharma/dyninst/patchAPI
-- Processing dependent target parseAPI...
-- Found dependency location /home/sharma/dyninst/parseAPI
-- Processing dependent target symtabAPI...
-- Found dependency location /home/sharma/dyninst/symtabAPI
-- Building dynC_API...
-- Processing dependent target dyninstAPI...
-- Found dependency location /home/sharma/dyninst/dyninstAPI
-- Configuring DyninstAPI_RT in /home/sharma/dyninst/dyninstAPI_RT
-- Configuring RT library
-- The C compiler identification is GNU 7.5.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- -- Input platform: x86_64-unknown-linux2.4
-- Set arch and platform based definitions
-- Options set
-- Set optimization flags
-- Check if the system is big endian
-- Searching 16 bit integer
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of unsigned short
-- Check size of unsigned short - done
-- Using unsigned short
-- Check if the system is big endian - little endian
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/cc
-- dyninstAPI RT library SOVERSION: 10.1
-- dyninstAPI RT library LIBVERSION: 10.1.0
-- Checking for 32-bit runtime library...
-- Performing Test CHECK_RT_LIB_32
-- Performing Test CHECK_RT_LIB_32 - Success
-- Disabling 32-bit runtime library; change BUILD_RTLIB_32 to ON and install 32-bit build environment to enable
-- Configuring done
-- Generating done
-- Build files have been written to: /home/sharma/dyninst/dyninstAPI_RT
-- RTlib Makefile: 
-- Could NOT find LATEX (missing: LATEX_COMPILER) 
-- LaTeX not found
-- Adding Unix-specific dependencies
-- Configuring incomplete, errors occurred!
See also "/home/sharma/dyninst/CMakeFiles/CMakeOutput.log".
See also "/home/sharma/dyninst/CMakeFiles/CMakeError.log".

**Comment by hainest** [comment]
Please follow the instructions from the issue template when submitting build errors. We need to know what your environment looks like in order to give you any useful information. Also, please use gists or attachments for submitting error logs. Posting them directly into an issue makes them hard to follow.

---

> Could NOT find LATEX (missing: LATEX_COMPILER)

You either need to install a LaTeX library or disable building the Dyninst documentation by passing -DBUILD_DOCS=NO to cmake.

>/usr/bin/ld: cannot find -lpthreads

I rather doubt pthreads is missing from your system. It's more likely that it is installed in a non-standard location. You can use `dpkg -L libc6 | grep -i pthread` to see where it is installed.

**Comment by hainest** [comment]
Closing due to inactivity. Please create a new issue if you are unable to resolve this issue.
